### PR TITLE
Admin assets and user

### PR DESCRIPTION
**Pull request by innatewonderer**

### CODE REVIEW

**Comment by sudara** [comment]
Looks like there's some master conflicts, just fyi!

Here are some additional items:

* [ ] Make a column on the left of the main content to add `<ul>` menu for admin. I'll ask @ofsound for best css recommendation.
* [ ] Remove "spam" true/false col
* [ ] Remove year from date.
* [ ] Do we need the ID column actually? Thinking about how to deprioritize, maybe putting it with the date on a second line.
* [ ] Only show spam/unspam depending on status
* [ ] Add user to the same column with IP so it's like `sudara (54.58.79.1)`
* [ ] Both user and ip should be links to add `params[:filter_by]`
  * [ ] When `filter_by` exists, controller adds either ip or user scope
* [ ] We also want to filter by spam / unspam — these can go as nav items under "Comments" and can add either `?spam=true/false` or you can reuse the `filter_by` param and do some detection in the controller.

alonetone 2018-11-29 19-43-32

**Comment by innatewonderer** [comment]
@sudara I addressed all of the feedback from our discussion and mentioned above ^. let's QA and merge this... Becoming too big of a PR.

File: app/javascript/controllers/admin/comment_controller.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    // console.log("Hello, Stimulus!", this.element)
+  }
+
+  spam() {
+    // this.spamvalueTarget
+    this.unspamTarget.style.display = 'block'
+    this.spamTarget.style.display = 'none'
+  }
+
+  unspam() {
```
[the remark this answers is not in the document]
Comment (innatewonderer) [reply app/javascript/controllers/admin/comment_controller.js]: but what if I don't want to???

[the remark this answers is not in the document]
Comment (sudara) [reply app/javascript/controllers/admin/comment_controller.js]: Hehe, in that case our "grade" goes down I guess! Boo hoo. I guess alternatives are either a `toggleDisplay` method that you can pass both targets into, or `showSpamTarget` and `showUnspamTarget` methods, which I might prefer.

File: app/views/layouts/white_theme.html.erb [hunk 2]
```diff
@@ -36,6 +36,7 @@
 
   <header id="site_header" data-turbolinks-permanent>
     <%= render partial: 'shared/header' %>
+    <%= render partial: 'shared/admin' %>
```
Comment (sudara) [inline app/views/layouts/white_theme.html.erb]: Oh, we probably don't want to keep this (or the partial) now that we have a side nav.

File: app/controllers/admin/assets_controller.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    def spam
+      @asset.spam!
+      @asset.update_column :is_spam, true
+      flash.notice = "Track was marked as spam"
+      redirect_back(fallback_location: root_path)
+    end
+
+    def mark_group_as_spam
+      binding.pry
```
[the remark this answers is not in the document]
Comment (sudara) [reply app/controllers/admin/assets_controller.rb]: @innatewonderer definitely don't wanna hang the web processes here :)

File: app/javascript/controllers/admin/comment_controller.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      showUnspamTarget()
+    }
+  }
+
+  showSpamTarget() {
+    this.unspamTarget.style.display = 'block'
+    this.spamTarget.style.display = 'none'
+  }
+
+  showUnspamTarget(){
```
[the remark this answers is not in the document]
Comment (innatewonderer) [reply app/javascript/controllers/admin/comment_controller.js]: Hmmmm... I don't actually see two methods defined by same name. It's called in the toggle method (which I'm unable to get to work with ajax:success) but that's a problem for another day  :)

[the remark this answers is not in the document]
Comment (sudara) [reply app/javascript/controllers/admin/comment_controller.js]: Maybe it's just complaining because of chunks of same code in initialize

**Pull request merged**